test/e2e/cli_test.go

```diff
@@ -659,6 +677,46 @@ func (s *CLIWithServerSuite) TestWorkflowRetry() {
 		})
 }
 
+func (s *CLIWithServerSuite) TestRetryByLabelSelector() {
+	s.testNeedsOffloading()
+	var retryTime corev1.Time
+
+	s.Given().
+		Workflow("@testdata/retry-test.yaml").
+		When().
+		SubmitWorkflow().
+		WaitForWorkflow(fixtures.ToStart).
+		WaitForWorkflow(fixtures.Condition(func(wf *wfv1.Workflow) bool {
+			return wf.Status.AnyActiveSuspendNode()
+		}), "suspended node").
+		RunCli([]string{"terminate", "--selector", "argo-e2e=true"}, func(t *testing.T, output string, err error) {
+			if assert.NoError(t, err) {
+				assert.Contains(t, output, "workflow retry-test terminated")
+			}
+		}).
+		WaitForWorkflow(fixtures.Condition(func(wf *wfv1.Workflow) bool {
+			retryTime = wf.Status.FinishedAt
+			return wf.Status.Phase == wfv1.NodeFailed
+		}), "is terminated", 20*time.Second).
+		RunCli([]string{"retry", "--selector", "argo-e2e=true", "--restart-successful", "--node-field-selector", "templateName==steps-inner"}, func(t *testing.T, output string, err error) {
+			if assert.NoError(t, err) {
+				assert.Contains(t, output, "Name:")
+				assert.Contains(t, output, "Namespace:")
+			}
+		}).
+		WaitForWorkflow(fixtures.Condition(func(wf *wfv1.Workflow) bool {
```

Comment:
can we extract this condition so we can re-use it? `fixtures.AnyActiveNodeSuspened`?